src/circuit.rs

```diff
@@ -745,6 +789,27 @@ impl plonk::Circuit<pallas::Base> for Circuit {
                     0,
                 )?;
 
+                region.assign_advice(
+                    || "split_flag",
+                    config.advices[8],
+                    0,
+                    || {
+                        self.split_flag
+                            .map(|split_flag| pallas::Base::from(split_flag as u64))
+                    },
+                )?;
+
+                nf_old
```

Comment:
why `nf_old.inner()` is of different type from `root` `v_old` and `v_new`?

Reply:
root, v_old and v_new are some AssignedCell
Whereas, nf_old is a X<Affine, EccChip> which contains an EccChip and an AssignedCell
nf_old.inner() returns the AssignedCell of nf_old